server/game/cards/02_SHD/units/FinalizerMightOfTheFirstOrder.ts

```diff
@@ -0,0 +1,62 @@
+import AbilityHelper from '../../../AbilityHelper';
+import * as Helpers from '../../../core/utils/Helpers.js';
+import { Card } from '../../../core/card/Card';
+import { TriggeredAbilityContext } from '../../../core/ability/TriggeredAbilityContext';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { RelativePlayer, TargetMode, WildcardCardType, WildcardZoneName, ZoneName } from '../../../core/Constants';
+
+export default class FinalizerMightOfTheFirstOrder extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '9752523457',
+            internalName: 'finalizer#might-of-the-first-order',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'Choose any number of friendly units',
+            targetResolver: {
+                activePromptTitle: 'Choose friendly units that will capture enemy units in the same arena',
+                mode: TargetMode.UpToVariable,
+                cardTypeFilter: WildcardCardType.Unit,
+                zoneFilter: WildcardZoneName.AnyArena,
+                controller: RelativePlayer.Self,
+                canChooseNoCards: true,
+                numCardsFunc: (context) => Math.min(
+                    context.source.controller.getUnitsInPlay().length,
+                    this.countOpponentNonLeaderUnitsInPlay(context, WildcardZoneName.AnyArena)
+                ),
+                multiSelectCardCondition: (card, selectedCards, context) => this.countOpponentNonLeaderUnitsInPlay(context, card.zoneName) > this.countSelectedCardsInZone(selectedCards, card.zoneName),
+            },
+            then: (chosenUnitsContext) => ({
+                title: 'Each of those units captures an enemy non-leader unit in the same arena',
+                immediateEffect: AbilityHelper.immediateEffects.sequential(
```

Comment:
I never understood if for something like this it makes a difference to use `sequential` vs `simultaneous`.

Reply:
Yeah there's definitely some cases where it doesn't really matter. The difference is mostly about whether constant abilities are given a chance to resolve in between effect - i.e., whether other cards "see" these things happen as separate effects or all at once.

In this case I think it should be all at once from the wording on the card, if that doesn't create issues with it working correctly. Like if there was some relevant Coordinate ability on the opponent side that was keeping a unit alive, with `sequential` the unit could potentially die to seeing Coordinate get shut off but with `simultaneous` it won't since it gets captured in the same moment as all of the others (which I'm pretty sure is the correct behavior here).

Reply:
Makes a lot of sense, thanks!`Huyang` could be another example of a card that could kill something if it's captured before the other.

I guess we would need `simultaneous` here but it's not as straight forward as updating this line because we would need to be able to prevent the user to capture the same card from multiple units (e.g. if you take the test as an example, Iden captures the Superlaser Tech, then for the Wampa you shouldn't be able to choose the Superlaser Tech again).

Is there anything similar already? I'll need to have a think for how to do this 🤔

Reply:
Ok, I think I figured it out. I can use the context's events to see which cards have been captured already and filter them in the `selectCard` step. Let me know if you have a better suggestion!